raft/leaderElection/leaderelection.go

```diff
@@ -11,38 +11,37 @@ import (
 
 // LeaderElection implements Raft leader election
 func LeaderElection(config types.Configuration, IP string) {
-	// for !findALeader(config, IP) {
-	findALeader(config, IP)
-	log.Println("No leader found, repeating election")
-	// }
-	log.Println("Leader found!")
+	types.ServerData[IP] = incrementTermAndBecomeCandidate(types.ServerData, IP)
+	log.Printf("%s incremented term to %d\n", IP, types.ServerData[IP].CurrentTerm)
+
+	rand.Seed(time.Now().UnixNano())
+	timer, duration := getTimer(150, 300)
+	log.Printf("%s set timeout at %d\n", IP, duration)
+
+	voteChan := make(chan int)
+	requestvotes.RequestVotes(config, IP, timer, voteChan)
+	votes := <-voteChan
+	log.Printf("Votes for %s is %d\n", IP, votes)
+
+	if majorityObtained(votes, config) {
+		log.Printf("%v is elected as a leader", IP)
+	}
 }
 
-func incrementTermAndBecomeCandidate(IP string) {
-	state := types.ServerData[IP]
+func incrementTermAndBecomeCandidate(ServerData map[string]*types.State, IP string) *types.State {
+	state := ServerData[IP]
 	state.CurrentTerm++
 	state.Name = "candidate"
-	types.ServerData[IP] = state
+	return state
```

Comment:
I dont think this needs to return this. We're working on the pointer anyway.

Reply:
Testing global variables is hard. It's a better practice anyway.